tools/terraform/provider/provider_test.go

```diff
@@ -0,0 +1,33 @@
+package provider
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+)
+
+// testAccProtoV6ProviderFactories are used to instantiate a provider during
+// acceptance testing. The factory function will be invoked for every Terraform
+// CLI command executed to create a provider server to which the CLI can
+// reattach.
+var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
+	"fleetdm": providerserver.NewProtocol6WithError(New("test")()),
+}
+
+func testAccPreCheck(t *testing.T) {
+	// You can add code here to run prior to any test case execution, for example assertions
+	// about the appropriate environment variables being set are common to see in a pre-check
+	// function.
+	apiKey := os.Getenv("FLEETDM_APIKEY")
+	if apiKey == "" {
+		t.Skip("FLEETDM_APIKEY not set")
+	}
+
+	// I don't like this, but I can't figure out how to pass the url otherwise...
+	url := os.Getenv("FLEETDM_URL")
+	if url == "" {
+		t.Skip("FLEETDM_URL not set")
+	}
```

Comment:
Do we have an idea for a test implementation here? Or does just that actually do something if not skipped?

Reply:
This is used in `teams_resource_test.go` as the `PreCheck` part of test cases. That file is the (sort of magical) way that resources are tested in the terraform framework.  This code is really just more of a helper to the resource testing.  

As for testing the provider itself, iirc I didn't see much in example code. It wouldn't be all that interesting, it's really just the Configure() call, and it's (partially) tested through the resource testing.